### PR TITLE
Bugfix/wjiang/gnu pointer allocation

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
This PR is addressing issue #148 and issue #127

### CODE REVIEW

File: ESMF/UFS/Aerosol_Cap.F90 [hunk 1]
```diff
@@ -311,14 +311,16 @@ subroutine ModelDataInitialize(model, rc)
     end if
 
     ! startup MAPL
-    allocate(is % wrap % maplCap, stat=stat)
+    allocate(is % wrap % maplCap, stat=stat, source = MAPL_Cap("MAPL Driver", aerosol_routine_SS, cap_options=maplCapOptions, rc=rc))
```
Comment (tclune) [inline ESMF/UFS/Aerosol_Cap.F90:314]: I don't think `stat` is the right variable to check here.   Should be the `rc` code from the previous call.     Please also check the other calls to the log error routines.

Comment (weiyuan-jiang) [reply ESMF/UFS/Aerosol_Cap.F90:314]: After this call, both stat and rc are checked.  It should be fine

**Review by tclune — APPROVED** [review]
"CMake" changes ok.

**Review by vbuchard — APPROVED** [review]
Looks good to me.

**Pull request merged**